Source/WebCore/dom/make_names.pl

```diff
@@ -247,70 +301,56 @@ sub defaultInterfaceName
 
 ### Parsing handlers
 
-sub valueForName
+sub nameAvoidingCppKeyword
 {
     my $name = shift;
-    my $value = $extensionAttrs{$name};
 
-    if (!$value) {
-        $value = $name;
-        $value =~ s/_/-/g;
-    }
+    my %cppKeywords = map { $_ => 1 } qw(char class default for operator switch template);
+    $name .= "_" if $cppKeywords{$name};
 
-    return $value;
+    return $name;
 }
 
-sub namesToStrings
+sub makeKey
 {
-    my $namesRef = shift;
-    my $stringsRef = shift;
-
-    my %names = %$namesRef;
+    my $namespace = shift;
+    my $localName = shift;
 
-    for my $name (keys %names) {
-        $stringsRef->{$name} = valueForName($name);
-    }
+    return "$namespace:$localName";
 }
 
 sub tagsHandler
 {
     my ($tag, $property, $value) = @_;
 
-    $tag =~ s/-/_/g;
+    my $tagKey = makeKey($parameters{namespace}, $tag);
```

Comment:
Isn't this more of an element key since we're talking about namespace + tag?

Reply:
Indeed, you are right. I think that implies we should rename %allTags to %allElements, and various other similar names.